File: src/main/java/com/zzk/infrastructure/ai/adapter/StreamChunk.java
package com.zzk.infrastructure.ai.adapter;

/**
 * 统一流式响应块
 * 
 * <p>
 * 将各厂商的不同响应格式适配为统一模型，
 * 使前端能够区分深度思考和正式回答。
 * 
 * <p>
 * 设计模式：适配器模式（Adapter Pattern）
 * - 将不兼容的接口转换为统一接口
 * - 解耦厂商格式与前端渲染逻辑
 * 
 * @author zzk
 * @since 1.0.0
 */
public record StreamChunk(
        ChunkType type,
        String content) {
    /**
     * 响应块类型
     */
    public enum ChunkType {
        /**
         * 深度思考/推理过程
         * <p>
         * 对应：
         * - OpenAI o1: reasoning_content
         * - DeepSeek R1: reasoning_content
         * - Claude: thinking_delta
         * - Gemini: thought
         */
        REASONING,

        /**
         * 正式回答内容
         */
        CONTENT,

        /**
         * 工具调用
         */
        TOOL_CALL,

        /**
         * 流结束标识
         */
        DONE
    }

    /**
     * 创建推理类型的块
     */
    public static StreamChunk reasoning(String content) {
        return new StreamChunk(ChunkType.REASONING, content);
    }

    /**
     * 创建内容类型的块
     */
    public static StreamChunk content(String content) {
        return new StreamChunk(ChunkType.CONTENT, content);
    }

    /**
     * 创建工具调用类型的块
     */
    public static StreamChunk toolCall(String content) {
        return new StreamChunk(ChunkType.TOOL_CALL, content);
    }

    /**
     * 创建结束标识块
     */
    public static StreamChunk done() {
        return new StreamChunk(ChunkType.DONE, null);
    }

    /**
     * 判断是否为有效内容块
     */
    public boolean hasContent() {
        return content != null && !content.isEmpty();
    }
}

File: go-ai-gateway/strategy/stream_chunk.go
package strategy

// ChunkType 响应块类型
type ChunkType string

const (
	// ChunkReasoning 深度思考/推理过程
	// OpenAI o1: reasoning_content
	// DeepSeek R1: reasoning_content / reasoning
	// Claude: thinking_delta
	// Gemini: thought
	ChunkReasoning ChunkType = "reasoning"

	// ChunkContent 正式回答内容
	ChunkContent ChunkType = "content"

	// ChunkToolCall 工具调用
	ChunkToolCall ChunkType = "tool_call"

	// ChunkDone 流结束标识
	ChunkDone ChunkType = "done"
)

// StreamChunk 统一流式响应块
// 适配器模式：将各厂商的不同响应格式适配为统一模型
type StreamChunk struct {
	Type    ChunkType `json:"type"`
	Content string    `json:"content"`
}

// Reasoning 创建推理类型的块
func Reasoning(content string) StreamChunk {
	return StreamChunk{Type: ChunkReasoning, Content: content}
}

// Content 创建内容类型的块
func Content(content string) StreamChunk {
	return StreamChunk{Type: ChunkContent, Content: content}
}

// ToolCall 创建工具调用类型的块
func ToolCall(content string) StreamChunk {
	return StreamChunk{Type: ChunkToolCall, Content: content}
}

// Done 创建结束标识块
func Done() StreamChunk {
	return StreamChunk{Type: ChunkDone}
}

// HasContent 判断是否为有效内容块
func (c StreamChunk) HasContent() bool {
	return c.Content != ""
}
